app/models/developer_external_profiles/linkedin_profile_fetcher.rb

```diff
@@ -0,0 +1,48 @@
+module DeveloperExternalProfiles
+  class LinkedinProfileFetcher
+    def developer_profiles
+      developer_external_profiles_list = []
+      developers_with_linked_in_profiles.each do |developer|
+        response = fetch_linkedin_profile(developer.linkedin)
+        external_profile_record = external_profile(developer, response)
+        developer_external_profiles_list << external_profile_record if external_profile_record.present?
+      end
+
+      upsert_external_profiles(developer_external_profiles_list)
+    end
+
+    def external_profile(developer, response)
+      if response[:error]
+        {developer_id: developer.id, site: "linkedin", data: {}, error: response[:error]}
+      else
+        external_profile = Developers::ExternalProfile.linkedin_developer(developer)
+        external_profile.data = response[:data] unless external_profile.blank?
+        if external_profile.blank? || external_profile.data_changed?
```

Comment:
I'm not sure I understand this logic. Why check for blank _or_ changed? Can you explain what's going on here?

Reply:
external_profile.blank? - This check is done to create new entry in external_profile table for the developer if there is no linkedin data associated with that developer already present in table
external_profile.data_changed? - This check is done to update existing entry of developer in external_profile table only if developer's latest company experience received from API is changed from the one already stored in database for it.

Basically, I am doing this check so that external_profile record is not updated if there are no changes as weekly_profile mail sends list of only those developers who are newly added or its experience has changed in last 7 days.

Reply:
Ah! That makes sense - thanks for explaining.